x-pack/plugins/transform/public/app/sections/create_transform/components/step_define/common/top_metrics_agg/components/top_metrics_agg_form.tsx

```diff
@@ -28,7 +28,7 @@ export const TopMetricsAggForm: PivotAggsConfigTopMetrics['AggFormComponent'] =
   } = useContext(PivotConfigurationContext)!;
 
   const sortFieldOptions = fields
-    .filter((v) => TOP_METRICS_SORT_FIELD_TYPES.includes(v.type))
+    .filter((v) => (TOP_METRICS_SORT_FIELD_TYPES as string[]).includes(v.type))
```

Comment:
It would be great if we could keep this unchanged without the `string[]` casting. The problem is that `TOP_METRICS_SORT_FIELD_TYPES` is inferred to only be `KBN_FIELD_TYPES[]`.  Suggest the following changes:

`x-pack/plugins/transform/public/app/sections/create_transform/components/step_define/common/types.ts`:
```
export type FieldType = KBN_FIELD_TYPES | TIME_SERIES_METRIC_TYPES.COUNTER;

export interface Field {
  name: EsFieldName;
  type: FieldType;
}
```

`x-pack/plugins/transform/public/app/common/pivot_aggs.ts`:
```
export const TOP_METRICS_SORT_FIELD_TYPES: FieldType[] = [
  KBN_FIELD_TYPES.NUMBER,
  KBN_FIELD_TYPES.DATE,
  KBN_FIELD_TYPES.GEO_POINT,
];
```

Reply:
I missed this previously but counter fields are a valid top metrics sort field type. Updated it here [`79a6591` (#154171)](https://github.com/elastic/kibana/pull/154171/commits/79a659138a929f0a5e515704a309d9de95e4304a)

Options should now show it as a sort field:
<img width="1792" alt="Screen Shot 2023-04-04 at 09 55 16" src="https://user-images.githubusercontent.com/43350163/229832799-551e5ac1-905d-47bd-aac8-8fda14104c39.png">